mesa/time.py

```diff
@@ -142,12 +142,12 @@ class SimultaneousActivation(BaseScheduler):
 
     def step(self) -> None:
         """Step all agents, then advance them."""
-        agent_keys = list(self._agents.keys())
+        agent_keys = self._agents.keys()
         for agent_key in agent_keys:
             self._agents[agent_key].step()
         # We recompute the keys because some agents might have been removed in
         # the previous loop.
-        agent_keys = list(self._agents.keys())
+        agent_keys = self._agents.keys()
```

Comment:
The comment above will become obsolete. We should change it to say that the previous step might remove some agents, but this loop will go over the remaining existing agents, and so, there is no problem.